WordPress/gutenberg
# Patterns: Apply layout and alignment to synced patterns in the editor

Fixes:
- https://github.com/WordPress/gutenberg/issues/8288

Related:
- https://github.com/WordPress/gutenberg/pull/54289
- https://github.com/WordPress/gutenberg/pull/54170

## What?

Adds alignment and layout to the editor-only `core/block` wrapper to maintain visual consistency when converting to a pattern.


## Why?

It's important to provide a smooth editing experience around patterns. Creating a pattern from wide, full, left, or right-aligned blocks, only to see them move and get squashed into a pattern block wrapper in the editor is unexpected.


## How?

- Only checks the pattern's inner blocks when they are first loaded so that if the pattern has wide/full/left/right aligned blocks, then even if those alignments are removed temporarily they can be re-applied
- If the `core/block` instance needs a layout applied it is inherited from the parent layout if available.
- Avoids the need to adopt layout support for the editor only by directly generating the appropriate layout styles
- The inferred alignment and layout, based on the pattern's inner blocks and the parent layout, are no longer set in the block's attributes avoiding any need to flag changes as not persisted
- Tweaks root padding styles to maintain consistent padding before and after converting blocks to a pattern



## Testing Instructions

See testing instructions on:
- https://github.com/WordPress/gutenberg/pull/54289
- https://github.com/WordPress/gutenberg/pull/54170
- https://github.com/WordPress/gutenberg/pull/54289#issuecomment-1717245813

## Screenshots or screencast
